elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/impl/ElasticsearchSourceStage.scala

```diff
@@ -109,7 +109,16 @@ private[elasticsearch] final class ElasticsearchSourceLogic[T](indexName: String
           }
         )
 
-        val completeParams = searchParams ++ extraParams.flatten
+        val queryParams = {
+          val baseMap = Map("scroll" -> settings.scroll, "sort" -> "_doc")
+          if (searchParams.contains("routing")) baseMap + ("routing" -> searchParams("routing"))
+          else baseMap
+        }
+
+        val completeParams = {
+          if (searchParams.contains("routing")) searchParams - "routing" ++ extraParams.flatten
+          else searchParams ++ extraParams.flatten
+        }
```

Comment:
Just a minor style comment. Unsure if you need to check multiple times for the presence of a `routing` key in the map. Personally think just doing something like:
```scala
val baseMap        = Map("scroll" -> settings.scroll, "sort" -> "_doc")
val routingKey     = "routing"
val queryParams    = searchParams.get(routingKey).fold(baseMap)(r => baseMap + (routingKey -> r))
val completeParams = searchParams ++ extraParams.flatten - routingKey
```
is a bit tidier but not a big issue.

Reply:
Great suggestion will take that into account.